pkg/api/query/v1.go

```diff
@@ -309,8 +309,9 @@ type queryData struct {
 	Result     parser.Value     `json:"result"`
 	Stats      stats.QueryStats `json:"stats,omitempty"`
 	// Additional Thanos Response field.
-	QueryAnalysis queryTelemetry `json:"analysis,omitempty"`
-	Warnings      []error        `json:"warnings,omitempty"`
+	QueryAnalysis      queryTelemetry             `json:"analysis,omitempty"`
+	Warnings           []error                    `json:"warnings,omitempty"`
+	SeriesStatsCounter storepb.SeriesStatsCounter `json:"seriesStatsCounter,omitempty"`
```

Comment:
I'm against this, it's a feature that could be so much more useful. For example, if you have a query like `sum(a) + sum(b)` then how do you know which selector downloaded more data and what should be the focus of optimizations? We could do much better. In fact, there's a mechanism/place where this kind of info should be exposed.

Please take a look at the Analyze functionality and focus there. A good plan of action would be first to add some mechanism through context.Context that would allow capturing more granular data about the fan-out. This could be used in the PromQL engine to "surface" this data through the Analyze() call, and then this could be shown in the UI next to each selector.

Reply:
Hi @GiedriusS  in our case our usecase is much simpler: we simply want to know how expensive each query is and attribute our total query cost to each user based on # of bytes fetched/# of timeseries. 
The method you mentioned is an overkill for this usecase and could be put as a TODO, what do you think?